app/src/main/java/com/android/streetworkapp/model/park/ParkViewModel.kt

```diff
@@ -102,6 +110,30 @@ open class ParkViewModel(private val repository: ParkRepository) : ViewModel() {
   fun updateName(pid: String, name: String) =
       viewModelScope.launch { repository.updateName(pid, name) }
 
+  /** Update the name of the current park using the nominatim API. */
+  fun updateCurrentParkNameNominatim() =
+      viewModelScope.launch {
+
+        // Only true when we have never updated the park name with nominatim to reduce calls to the
+        // API
+        if (_currentPark.value?.name?.contains("Default Park") == true) {
+          nameRepository.convertLocationIdToParkName(
+              _parkLocation.value.id,
+              { name -> _currentPark.value?.let { repository.updateName(it.pid, name) } },
+              {})
```

Comment:
Logging in case of an error could be a good idea.

Reply:
You are right I will update it, thank you!